proctor-common/src/main/java/com/indeed/proctor/common/TestRangeSelector.java

```diff
@@ -37,6 +37,7 @@ public class TestRangeSelector {
     @Nonnull private final ConsumableTestDefinition testDefinition;
     @Nonnull private final String[] rules;
     @Nonnull private final TestBucket[][] rangeToBucket;
+    private String currentRule;
```

Comment:
An alternative would be to format the error thrown by: evaluator.apply to include the rule instead of catching and throwing in the findMatchingRuleInternal method. I will look into doing it this way.

Reply:
Even simpler alternative check for excpetion in the getMatchingAllocation which tests the allocation level ruels. Will fix this.